### DISCUSSION TITLE
improving the PageChooserBlock to have the options of richtext text linking

### DISCUSSION THREAD

**Discussion by hazho** [wagtail/wagtail#7734] [Ideas]
![image](https://user-images.githubusercontent.com/20362375/143544266-36cd5afc-05a8-4a86-b456-1172377c0c7b.png)
this is the normal PageChooserBlock modal, the better approach is the one like richtext text linking (see bellow):
![image](https://user-images.githubusercontent.com/20362375/143544478-6eb9af86-f223-4c71-b87c-0e9cd99c5088.png)
before I make a new issue with new feature flag, is there any problems that may prevent the wagtail core team to apply it?

**Comment by gasman** [comment]
Thanks for the suggestion!
I don't think this could be used directly in PageChooserBlock, since there are often situations where the chosen page is used in ways where a simple external link wouldn't work. For example, you might use it to choose a "featured blog post" on a listing page, where the intro paragraph and lead image from that blog post get pulled out to be shown within the listing. It could certainly work as a new block type, though.

There's an existing ticket at #381 covering this subject - doing this within a StreamField is a bit simpler than the general case of a unified link chooser, since we don't have to deal with the different database representations for page IDs versus external URLs.
